gum/backend-x86/gumstalker-x86.c

```diff
@@ -4625,91 +4646,114 @@ gum_exec_block_is_direct_jmp_to_plt_got (GumExecBlock * block,
                                          GumGeneratorContext * gc,
                                          GumBranchTarget * target)
 {
-  gboolean result = FALSE;
-
 #if defined (HAVE_LINUX)
   GumExecCtx * ctx = block->ctx;
   const cs_insn * insn = gc->instruction->ci;
-  GumModuleMap * map = NULL;
-  const GumModuleDetails * module;
-  GumElfModule * elf = NULL;
-  GumCheckElfSection plt_got = {
-    .name = ".plt.got",
-    .target = target,
-    .found = FALSE
-  };
-  GumCheckElfSection plt_sec = {
-    .name = ".plt.sec",
-    .target = target,
-    .found = FALSE
-  };
+  GArray * ranges;
+  GumMemoryRange *range;
 
   if (target->is_indirect)
-    goto beach;
+    return FALSE;
 
   if (target->base != X86_REG_INVALID)
-    goto beach;
+    return FALSE;
 
   if (ctx->activation_target != NULL)
-    goto beach;
+    return FALSE;
 
   if (!gum_stalker_is_excluding (ctx->stalker, target->absolute_address))
-    goto beach;
+    return FALSE;
 
   if (insn->id != X86_INS_JMP)
-    goto beach;
+    return FALSE;
 
-  map = gum_module_map_new ();
+  ranges = gum_exec_ctx_get_plt_got_ranges ();
 
-  module = gum_module_map_find (map, GUM_ADDRESS (target->absolute_address));
-  if (module == NULL)
-    goto beach;
+  for (guint i = 0; i < ranges->len; i++) {
+    range = &g_array_index(ranges, GumMemoryRange, i);
+
+    if (GUM_MEMORY_RANGE_INCLUDES (range,
+        GPOINTER_TO_SIZE (target->absolute_address)))
```

Comment:
Perhaps we should `g_array_sort()` the ranges, and do a binary search using `g_array_binary_search()`?

Reply:
The array should be quite small and only used on the slow path.